manim/mobject/types/vectorized_mobject.py

```diff
@@ -264,13 +267,12 @@ def set_stroke(
             array_name = "stroke_rgbas"
             width_name = "stroke_width"
             opacity_name = "stroke_opacity"
-            color_name = "stroke_color"
         self.update_rgbas_array(array_name, color, opacity)
         if width is not None:
             setattr(self, width_name, width)
         if opacity is not None:
             setattr(self, opacity_name, opacity)
-        if color is not None:
+        if background:
```

Comment:
Why switch this to depending on `background`? Also, it might be better to have it lumped in with the rest of the `if background: ` attribute setting.

Reply:
The old condition, `color is not None`, should also be present, I've fixed it now. It is depending on `background` because only then the `color_name` should be updated, otherwise it does not exist. I've made it more explicit, so hopefully the code is more understandable now.